test/facility-guarantee/post-facility-guarantee.api-test.ts

```diff
@@ -0,0 +1,302 @@
+
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+  const lenderTypeCode = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.lenderType.lenderTypeCode;
+  const limitTypeCode = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.limitType.limitTypeCode;
+  const sectionIdentifier = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.sectionIdentifier;
+  const guaranteedPercentage = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.guaranteedPercentage;
+
+  const guarantorParty = valueGenerator.stringOfNumericCharacters({ length: 8 });
+  const limitKey = valueGenerator.stringOfNumericCharacters({ length: 8 });
+  const effectiveDateInPast = TEST_DATES.A_PAST_EFFECTIVE_DATE_ONLY;
+  const guaranteeExpiryDateInFuture = TEST_DATES.A_FUTURE_EXPIRY_DATE_ONLY;
+  const guaranteeTypeCode = valueGenerator.stringOfNumericCharacters({ length: 3 });
+  const maximumLiability = 12345.6;
+
+  const acbsRequestBodyToCreateFacilityGuarantee = {
+    LenderType: {
+      LenderTypeCode: lenderTypeCode,
+    },
+    SectionIdentifier: sectionIdentifier,
+    LimitType: {
+      LimitTypeCode: limitTypeCode,
+    },
+    LimitKey: limitKey,
+    GuarantorParty: {
+      PartyIdentifier: guarantorParty,
+    },
+    GuaranteeType: {
+      GuaranteeTypeCode: guaranteeTypeCode,
+    },
+    EffectiveDate: dateStringTransformations.addTimeToDateOnlyString(effectiveDateInPast),
+    ExpirationDate: dateStringTransformations.addTimeToDateOnlyString(guaranteeExpiryDateInFuture),
+    GuaranteedLimit: 12345.6,
+    GuaranteedPercentage: guaranteedPercentage,
+  };
+
+  const requestBodyToCreateFacilityGuarantee = [
+    {
+      facilityIdentifier,
+      guarantorParty,
+      limitKey,
+      effectiveDate: effectiveDateInPast,
+      guaranteeExpiryDate: guaranteeExpiryDateInFuture,
+      maximumLiability,
+      guaranteeTypeCode,
+    },
+  ];
+
+  let api: Api;
+
+  beforeAll(async () => {
+    api = await Api.create();
+  });
+
+  afterAll(async () => {
+    await api.destroy();
+  });
+
+  afterEach(() => {
+    nock.abortPendingRequests();
+    nock.cleanAll();
+  });
+
+  const { idToken, givenAuthenticationWithTheIdpSucceeds } = withAcbsAuthenticationApiTests({
+    givenRequestWouldOtherwiseSucceed: () => givenRequestToCreateFacilityGuaranteeInAcbsSucceeds(),
+    makeRequest: () => api.post(createFacilityGuaranteeUrl, requestBodyToCreateFacilityGuarantee),
+    successStatusCode: 201,
+  });
+
+  withClientAuthenticationTests({
+    givenTheRequestWouldOtherwiseSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenRequestToCreateFacilityGuaranteeInAcbsSucceeds();
+    },
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) =>
+      api.postWithoutAuth(createFacilityGuaranteeUrl, requestBodyToCreateFacilityGuarantee, incorrectAuth?.headerName, incorrectAuth?.headerValue),
+  });
+
+  it('returns a 201 response with the facility guarantee location if it has been successfully created in ACBS', async () => {
```

Comment:
Is this description clear/accurate? Would `'returns a 201 response with the facility identifier if the guarantee was successfully created in ACBS'` be clearer?